README.md

```diff
@@ -175,7 +175,7 @@ public interface Subscription {
 | [:bulb:](#3.3 "3.3 explained") | *The intent of this rule is to complement [see [3.2](#3.2)] by placing an upper limit on the mutual recursion between `request` and `onNext` (and eventually `onComplete` / `onError`). Implementations are RECOMMENDED to limit this mutual recursion to a depth of `1` (ONE)—for the sake of conserving stack space. An example for undesirable synchronous, open recursion would be Subscriber.onNext -> Subscription.request -> Subscriber.onNext -> …, as it otherwise will result in blowing the calling thread´s stack.* |
 | <a name="3.4">4</a>       | `Subscription.request` SHOULD respect the responsivity of its caller by returning in a timely manner. |
 | [:bulb:](#3.4 "3.4 explained") | *The intent of this rule is to establish that `request` is intended to be a [non-obstructing](#term_non-obstructing) method, and should be as quick to execute as possible on the calling thread, so avoid heavy computations and other things that would stall the caller´s thread of execution.* |
-| <a name="3.5">5</a>       | `Subscription.cancel` MUST respect the responsivity of its caller by returning in a timely manner, MUST be idempotent and MUST be [thread-safe](#term_thread-safe). |
+| <a name="3.5">5</a>       | `Subscription.cancel` MUST respect the responsivity of its caller by returning in a timely manner, MUST be idempotent and MUST be executed [serially](#term_serially). |

```

Comment:
@davidmoten @reactive-streams/contributors @OlegDokuka 

Should this read: `MUST be executed [serially](#term_serially)`?

Reply:
Ok, now I'm thinking that is redundant since it overlaps with 2.7

Reply:
fixed. Removed `and MUST be executed [serially](#term_serially)` since it is the responsibility of `Subscriber` to do make call serially on `Subscription` 

/cc @simonbasle @smaldini @akarnokd

Reply:
No, do not remove that sentence please.

Reply:
ok, then we need to rever back thread-safe notion

Reply:
but it will contradict with serial access because thread-safety does not require external sync where 2.7 states that external sync is needed on calling cancel. Thus it seems that serial access needed only for `Subscription#request`

Reply:
In general, the 2.7 had sounded like `access to your stored subscription in the way you get access to the field from any thread without a problem`. Thus external synchronization was mentioned in order to clarify that calling to subscription should be with happens before. It did not say that calling to methods must be done serially

Reply:
To convey that the implementation of cancel needs to be thread safe, the following verbiage should work: "and MUST be implemented to execute [serially](#term_serially)"?